pyccel/parser/semantic.py

```diff
@@ -739,7 +739,10 @@ def _infer_type(self, expr):
             d_var['cls_base'       ] = TupleClass
             d_var['memory_handling'] = 'heap'
             return d_var
-
+        elif isinstance(expr, PythonList):
+            d_var['cls_base'       ] = ListClass
+            d_var['memory_handling'] = 'heap'
+            return d_var
```

Comment:
Is this condition necessary? Can you explain why this is not handled by the condition below `elif isinstance(expr, TypedAstNode):`? (Maybe `PythonTuple` caused confusion? The condition `elif isinstance(expr, PythonList):` is removed in one of my unfinished PRs as it is no longer necessary)

Reply:
i guess you're right, it's not needed

Reply:
Fixed ad6fca5eb78d573c514430038008ad187d865733